DataFormats/HcalRecHit/interface/HFQIE10Info.h

```diff
@@ -0,0 +1,40 @@
+#ifndef DATAFORMATS_HCALRECHIT_HFQIE10INFO_H
+#define DATAFORMATS_HCALRECHIT_HFQIE10INFO_H
+
+#include "DataFormats/HcalDetId/interface/HcalDetId.h"
+
+/** \class HFQIE10Info
+*
+* Class to contain the info needed to perform HF reconstruction
+* using QIE10 chips and dual-anode readout. Intended for use
+* inside HFPreRecHit.
+*/
+class HFQIE10Info
+{
+public:
+    typedef HcalDetId key_type;
+
+    HFQIE10Info();
+
+    HFQIE10Info(const HcalDetId& id, int adc, float charge,
+                float energy, float timeRising, float timeFalling);
+
+    inline HcalDetId id() const {return id_;}
+
+    inline int adc() const {return adc_;}
+    inline float charge() const {return charge_;}
+    inline float energy() const {return energy_;}
+    inline float timeRising() const {return timeRising_;}
+    inline float timeFalling() const {return timeFalling_;}
+
+private:
+    HcalDetId id_;
+
+    int32_t adc_;
+    float charge_;
+    float energy_;
+    float timeRising_;
+    float timeFalling_;
```

Comment:
Yes, I am aware of that. In the future, we might need
to have some calibration for times. In this case, it is
better to represent the times by floats.

On 05/05/2016 09:29 AM, Kevin Pedro wrote:
>
> In DataFormats/HcalRecHit/interface/HFQIE10Info.h <https://github.com/cms-sw/cmssw/pull/14383#discussion_r62194591>:
>
> > +    inline HcalDetId id() const {return id_;}
> > +
> > +    inline int adc() const {return adc_;}
> > +    inline float charge() const {return charge_;}
> > +    inline float energy() const {return energy_;}
> > +    inline float timeRising() const {return timeRising_;}
> > +    inline float timeFalling() const {return timeFalling_;}
> > +
> > +private:
> > +    HcalDetId id_;
> > +
> > +    int32_t adc_;
> > +    float charge_;
> > +    float energy_;
> > +    float timeRising_;
> > +    float timeFalling_;
>
> The TDC values are |int|s in the QIE10DataFrame <https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/DataFormats/HcalDigi/interface/QIE10DataFrame.h#L33> class.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/cms-sw/cmssw/pull/14383/files/d13235f9c646cc3c33425064ddf2551a81a4bef7#r62194591>
>